packages/itwin/grouping-mapping-widget/src/widget/components/QueryBuilder.ts

```diff
@@ -68,6 +91,33 @@ export class QueryBuilder {
       return false;
     }
 
+    const propertyField = (await this.dataProvider?.getFieldByPropertyRecord(
+      prop,
+    )) as PropertiesField;
+
+    if (!propertyField) {
+      toaster.negative(
+        "Error. Failed to fetch field for this property record.",
+      );
+      return false;
+    }
+
+    this._propertyMap.set(prop.property.name, { propertyRecord: prop, propertiesField: propertyField });
+    return true;
+  }
+
+  public removeProperty(prop: PropertyRecord) {
+    this._propertyMap.delete(prop.property.name);
+  }
+
+  private async buildProperty(prop: PropertyRecord, propertiesField: PropertiesField) {
```

Comment:
I don't think the `async` declaration is needed here.

Reply:
Removed async

Reply:
Looks like the only `async` methods left are `addProperty` and `removeProperty`.